### PR TITLE
Initial snap project with stack structure

### PR DESCRIPTION
**Pull request by ibarrae**
- Updated readme.me with installation steps.
- Created the snap project and then updated it to follow the stack project structure

### CODE REVIEW

File: src/Lib.hs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module Lib
+    ( someFunc
+    ) where
+
+someFunc :: IO ()
+someFunc = putStrLn "someFunc"
```
Comment (dbalseiro) [inline src/Lib.hs]: This file is redundant?

File: stack.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# resolver: nightly-2015-09-21
+# resolver: ghc-7.10.2
+# resolver: ghcjs-0.1.0_ghc-7.10.2
+#
+# The location of a snapshot can be provided as a file or url. Stack assumes
+# a snapshot provided as a file might change, whereas a url resource does not.
+#
+# resolver: ./custom-snapshot.yaml
+# resolver: https://example.com/snapshots/2018-01-01.yaml
+resolver: lts-11.17
```
Comment (dbalseiro) [inline stack.yaml]: use the newest lts please
https://www.stackage.org/lts-12.1

File: stack.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# This file was automatically generated by 'stack init'
```
Comment (dbalseiro) [inline stack.yaml]: can you clean up the commnets please?

File: snap.cabal.orig [hunk 4]
```diff
@@ -0,0 +1,28 @@
+Name:                snap
```
Comment (dbalseiro) [inline snap.cabal.orig:1]: What's the purpose of this file?

**Review by dbalseiro — CHANGES_REQUESTED** [review]
overall looks good... left a couple of comments

File: ChangeLog.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Changelog for snap
+1.0.1
+    - Added a package.yaml for the project.
+    - Untracked the .ghci file.
+    - Upgraded to lts-12.1
+    - Removed unnecesary comments on stack.yaml
+1.0
+    - Added a change log.
+    - Set up the initial folder structure for a snap project.
+    - Set up project structure for a 'stack' structure.
```
Comment (CristhianMotoche) [inline ChangeLog.md]: I suppose you want to display a list here, but the generated markdown doesn't look quite right. Perhaps, you want to set the `1.0` and `1.0.1` like subtitles:
```markdown
## 1.0.1
- Added a ...
```

File: snap.cabal.orig (hunk 4 above)
Comment (ibarrae) [reply snap.cabal.orig:1]: @dbalseiro Mainly for learning purposes, how the changes should end when comparing the initial file from the scratch with the final result.

File: ChangeLog.md (hunk 5 above)
Comment (CristhianMotoche) [reply ChangeLog.md]: @kinorius123 In addition to that. Why the version is `1.0.1` but the `package.yml` says `0.1.0.0`? Usually, you add things to your CHANGELOG accordingly with the version described in your cabal file (which is generated from `package.yml`).
Is that clear?

**Review by dbalseiro — APPROVED** [review]
:shipit: :clap:

**Pull request merged**